### PR TITLE
added test

### PR DESCRIPTION
**Pull request by alexandra-protyanova**
Screenshot 2024-01-22 at 15 31 16

### CODE REVIEW

File: tests/moveFiles.test.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  });
+
+  test('should throw error if destination directory does not exist', async() => {
+    const nonExistingDir = path.join(tempDir, 'nonExistingDir', 'test.txt');
+
+    const { stderr } = await execAsync(
+      `${basePath} ${testFile} ${nonExistingDir}`
+    );
+
+    expect(stderr).toContain('no such file or directory');
```
[the remark this answers is not in the document]
Comment (alexandra-protyanova) [reply tests/moveFiles.test.js:61]: I think, it is better to check, that error is present

File: tests/moveFiles.test.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    fs.mkdirSync(testDir);
+
+    const newPath = path.join(testDir, '/');
+
+    await execAsync(`${basePath} ${testFile} ${newPath}`);
+
+    expect(fs.existsSync(path.join(newPath, 'test.txt'))).toBe(true);
+  });
+
+  test('should move file into existing directory if last segment is directory', async() => {
```
[the remark this answers is not in the document]
Comment (alexandra-protyanova) [reply tests/moveFiles.test.js:75]: not actually, prev one was about inishing with a `/`, this one about directory name

**Comment by alexandra-protyanova** [comment]
destination is another existing folder + new filename (success, moved with the new filename)
destination is another existing folder + new filename without extension (success, moved with the new filename)

as for me it is overkill, so as if we have these passing test:
destination is the same filename only (success, not changed)
destination is the filename only without extension (success, moved)
destination is another existing folder (success, moved with the same filename)
destination is another non-existing folder (error)

2 cases aove gonna be fully covered)

thanks for your suggestions) cc @sergii-nosachenko

File: tests/moveFiles.test.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    fs.mkdirSync(testDir);
+
+    await execAsync(`${basePath} ${testFile} ${testDir}`);
+
+    expect(fs.existsSync(path.join(testDir, 'test.txt'))).toBe(true);
+  });
+
+  test('should rename file if last segment is not an existing directory', async() => {
```
[the remark this answers is not in the document]
Comment (alexandra-protyanova) [reply tests/moveFiles.test.js:83]: renamed)

File: tests/moveFiles.test.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  let testContent = '';
+  const testFileName = faker.system.commonFileName('txt');
+
+  const tempDir = path.join('tests', faker.word.noun());
+  const testFile = path.join(tempDir, testFileName);
+  const testDir = path.join('tests', faker.word.noun());
+
+  beforeAll(() => {
+    testContent = faker.lorem.paragraphs();
+  });
```
Comment (mgrinko) [inline tests/moveFiles.test.js:24]: why do you do it in `beforeAll` unlike the other constants?

File: tests/moveFiles.test.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+      const newFilePath = path.join(tempDir, faker.lorem.word());
+
+      const { stderr } = await execAsync(`${basePath} ${testFile} ${newFilePath}`);
+
+      expect(stderr).toBeFalsy();
+
+      const content = fs.readFileSync(newFilePath, 'utf-8');
+
+      expect(fs.existsSync(newFilePath)).toBe(true);
+      expect(fs.existsSync(testFile)).toBe(false);
```
Comment (mgrinko) [inline tests/moveFiles.test.js:71]: Why do you call the variables `testFile` but `newFilePath`?

File: tests/moveFiles.test.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+      expect(stderr).toBeFalsy();
+      expect(fs.existsSync(testFile)).toBe(true);
+      expect(content).toBe(testContent);
+    });
+
+    test('should move file, if passed destination is a file without extension', async() => {
+      const newFilePath = path.join(tempDir, faker.lorem.word());
+
+      const { stderr } = await execAsync(`${basePath} ${testFile} ${newFilePath}`);
```
Comment (mgrinko) [inline tests/moveFiles.test.js:88]:
```suggestion
      const newFilePath = path.join(tempDir, faker.lorem.word());
      const { stderr } = await execAsync(`${basePath} ${testFile} ${newFilePath}`);
```

File: tests/moveFiles.test.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    test('should move file, if passed destination is a directory', async() => {
+      fs.mkdirSync(testDir);
+
+      const { stderr } = await execAsync(`${basePath} ${testFile} ${testDir}`);
+
+      expect(stderr).toBeFalsy();
+
+      const newPath = path.join(testDir, testFileName);
+
+      const content = fs.readFileSync(newPath, 'utf-8');
```
Comment (mgrinko) [inline tests/moveFiles.test.js:104]:
```suggestion
      const newPath = path.join(testDir, testFileName);
      const content = fs.readFileSync(newPath, 'utf-8');
```

File: tests/moveFiles.test.js [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+      const { stderr } = await execAsync(
+        `${basePath} ${testFile} ${nonExistingDir}`
+      );
+
+      expect(stderr.length).toBeGreaterThan(0);
+      expect(fs.existsSync(nonExistingDir)).toBe(false);
+      expect(fs.existsSync(testFile)).toBe(true);
+    });
+
+    test('should throw error if destination is non-existed directory with fileName', async() => {
```
Comment (mgrinko) [inline tests/moveFiles.test.js:123]: Don't see any difference with the prev test

File: tests/moveFiles.test.js [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+      fs.mkdirSync(testDir);
+
+      const newPath = path.join(testDir, '/');
+
+      await execAsync(`${basePath} ${testFile} ${newPath}`);
+
+      expect(fs.existsSync(path.join(newPath, testFileName))).toBe(true);
+    });
+
+    test('should move file into existing directory if destination is a directory', async() => {
```
Comment (mgrinko) [inline tests/moveFiles.test.js:145]: What's the reason of this test if we alredy have another similar one
```
test('should move file, if passed destination is a directory'
```

**Review by mgrinko — APPROVED** [review]

**Pull request merged**